**Тузова Анастасия**

**ДЗ2**

**Задача 1.** 

Из партии батареек с номинальным напряжением 12 В наугад выбрано семь штук. Согласно показаниям высокоточного вольтметра, фактически выдаваемое ими напряжение составило: 12,9; 11,6; 13,5; 13,9; 12,1; 11,9; 13,0 В соответственно. Можно ли считать, что среднее напряжение действительно равно 12 В?

In [111]:
import numpy as np
from scipy import stats

Гипотеза о среднем <br>
$ Ho:  \mu_{\text{}} = \mu_{\text{0}}  $ <br>
$ Ha:  \mu_{\text{}} \neq \mu_{\text{0}} $ <br>

In [112]:
data = np.array([12.9, 11.6, 13.5, 13.9, 12.1, 11.9, 13.0])
m0 = 12

df = len(data) - 1 
t = stats.t(df) 
T = (data.mean() - m0) * (df)**0.5 / data.std()
p_value = min(2 * t.cdf(T), 2 - 2 * t.cdf(T))
print(f"Статистика критерия: {T}")
print(f"p-value: {p_value}")

Статистика критерия: 2.1578017974978794
p-value: 0.07430190210562304


На уровне значимости больше 0.07430190210562304: нулевая гипотеза отвергается --> значение среднего напряжения статистически отличается от 12 В. <br>
На уровне значимости меньше 0.07430190210562304: нулевая гипотеза не может быть отвергнута --> значение среднего напряжения статистически не отличается от 12 В. 

In [113]:
stats.ttest_1samp(data, m0)

TtestResult(statistic=np.float64(2.1578017974978794), pvalue=np.float64(0.07430190210562303), df=np.int64(6))

**Задача 2.**

Есть два различных метода определения процентного содержания жира в мясе. Доля жира для 10 образцов мяса была определена каждым из двух методов. Позволяют ли полученные результаты сделать вывод о том, что один метод даёт в среднем более высокие показания, чем другой?

Гипотеза о среднем при парных наблюдениях <br>
$ Ho:  \mu_{\text{}} = \mu_{\text{0}}, \mu_{\text{}} = \mu_{\text{x}} - \mu_{\text{y}} $ <br> 
$ Ha:  \mu_{\text{}} \neq \mu_{\text{0}} $ <br>

In [114]:
method1 = np.array([23.1, 23.2, 26.5, 26.6, 27.1, 48.3, 40.5, 25.0, 38.4, 23.5])
method2 = np.array([22.7, 23.6, 27.1, 27.4, 27.4, 46.8, 40.4, 24.9, 38.1, 23.8])
data = method1 - method2 
print(data)
m0 = 0

[ 0.4 -0.4 -0.6 -0.8 -0.3  1.5  0.1  0.1  0.3 -0.3]


In [115]:
df = len(data) - 1 
t = stats.t(df)
T = (data.mean() - m0) * (df)**0.5 / data.std()
p_value = min(2 * t.cdf(T), 2 - 2 * t.cdf(T))
print(f"Статистика критерия: {T}")  
print(f"p-value: {p_value}")

Статистика критерия: 1.715488497867803e-15
p-value: 0.9999999999999987


На уровне значимости меньше 0.9999999999999987 нулевая гипотеза не может быть отвергнута --> разница между двумя методами статистически не значима. 

In [116]:
stats.ttest_1samp(data, m0)

TtestResult(statistic=np.float64(1.7154884978678027e-15), pvalue=np.float64(0.9999999999999987), df=np.int64(9))

**Задача 3.** 

Для проверки гипотезы о том, что жители городов имеют в среднем более высокое систолическое артериальное давление, чем жители сельской местности, было проведено обследование 10 горожан и 9 жителей села примерно одного возраста. Получены следующие данные об уровне давления для горожан: 132, 111, 119, 138, 200, 131, 138, 170, 159, 140; для селян: 115, 190, 127, 155, 148, 121, 116, 121, 197. Можно ли считать верной выдвинутую гипотезу на уровне значимости 0,05?

Гипотеза о равенстве средних для независимых выборок <br>
$ Ho:  \mu_{\text{city}} = \mu_{\text{village}}  $ <br>
$ Ha:  \mu_{\text{city}} > \mu_{\text{village}} $ <br>

In [117]:
cityData = np.array([132, 111, 119, 138, 200, 131, 138, 170, 159, 140])
villageData = np.array([115, 190, 127, 155, 148, 121, 116, 121, 197])
m = len(cityData)
n = len(villageData)
s2city = cityData.var()
s2village = villageData.var()

In [118]:
T = (cityData.mean() - villageData.mean()) / (s2city / m + s2village / n)**0.5
df = round((s2city / m + s2village / n)**2 / ( (s2city / m)**2 / (m - 1) + (s2village / n)**2 / (n - 1))) 
t = stats.t(df)
p_value = 1 - t.cdf(T)
print(f"Статистика критерия: {T}")
print(f"p-value: {p_value}")

alpha = 0.05
if alpha > p_value:
    print("Отвергаем нулевую гипотезу на уровне значимости 0,05.")
else:
    print("Не можем отвергнуть нулевую гипотезу на уровне значимости 0,05.")

Статистика критерия: 0.03681832098940068
p-value: 0.48554266145506586
Не можем отвергнуть нулевую гипотезу на уровне значимости 0,05.


Разница между средними значениями систолического давления для городских и сельских жителей статистически не значима. 

In [119]:
stats.ttest_ind(cityData, villageData).pvalue / 2

np.float64(0.486178812200461)

**Задача 4.** 

Известно, что 52% новорождённых детей — мальчики, а 48% — девочки. В случайной выборке из 5 000 человек в возрасте от 30 до 60 лет мужчин оказалось 2 500. Можно ли на уровне значимости 0,05 считать, основываясь на этих данных, что смертность среди мужчин и женщин одинакова? 

Гипотеза о доле (вероятности) <br>
смертность одинаковая --> в случайной выборке из 5000 человек должно быть 52% мужчин <br>
$ Ho:  p = p_{\text{0}} $ <br>
$ Ha:  p \neq p_{\text{0}} $ <br>

In [120]:
n = 5000
m = 2500
p0 = 0.52

T = (m - n * p0) / (n * p0 * (1 - p0))**0.5
norm_dist = stats.norm(0, 1) 
p_value = min(2 * norm_dist.cdf(T), 2 - 2 * norm_dist.cdf(T))
print(f"Статистика критерия: {T}")
print(f"p-value: {p_value}")

alpha = 0.05
if alpha > p_value:
    print("Отвергаем нулевую гипотезу на уровне значимости 0,05.")
else:
    print("Не можем отвергнуть нулевую гипотезу на уровне значимости 0,05.")

Статистика критерия: -2.830692585361489
p-value: 0.004644733942292514
Отвергаем нулевую гипотезу на уровне значимости 0,05.


На уровне значимости 0,05 нулевая гипотеза отвергается --> разница в смертности мужчин и женщин статистически значима.

In [121]:
result = stats.binomtest(m, n, p0)
print(f"Статистика критерия: {result.statistic}")
print(f"p-value: {result.pvalue}")

Статистика критерия: 0.5
p-value: 0.004849502159545284


**Задача 5.** 

В первом эксперименте для сравнения двух средних X1=17, X2=16 применялся t-критерий Стьюдента, и эти различия оказались значимы (p = 0, 001).
Во втором исследовании, также при помощи t - критерия, сравнивались два средних X1=17, X2=36, и эти различия не значимы при p = 0,8. В чем может быть причина таких результатов, при условии, что нулевая и альтернативная гипотезы в обоих случаях одинаковы?

Возможные причины таких результатов:
1. Разница в дисперсии данных: в первом эксперименте дисперсия могла быть значительно меньше, из-за чего были выявлены значимые различия.
2. Разный размер выборок: слишком малая выборка во втором эксперименте или слишком большая в первом. 